tests/python/operators/fermi_hubbard_test.py

```diff
@@ -0,0 +1,251 @@
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for Fermi-Hubbard model Hamiltonian."""
+
+from __future__ import annotations
+
+import numpy as np
+import scipy
+
+import ffsim
+from ffsim._lib import FermionOperator
+from ffsim.operators.fermi_hubbard import fermi_hubbard_1d
+from ffsim.operators.fermion_action import cre_a, cre_b, des_a, des_b
+
+
+def test_non_interacting_fermi_hubbard_1d():
+    """Test non-interacting one-dimensional Fermi-Hubbard model Hamiltonian."""
+
+    # open boundary conditions
+    op = fermi_hubbard_1d(norb=4, tunneling=1, interaction=0)
+    ham = ffsim.linear_operator(op, norb=4, nelec=(2, 2))
+    coeffs: dict[tuple[tuple[bool, bool, int], ...], complex] = {}
+    for p in range(3):
+        coeffs[(cre_a(p), des_a((p + 1) % 4))] = -1
+        coeffs[(cre_b(p), des_b((p + 1) % 4))] = -1
+        coeffs[(cre_a((p + 1) % 4), des_a(p))] = -1
+        coeffs[(cre_b((p + 1) % 4), des_b(p))] = -1
```

Comment:
I had something a bit different in my mind for the test, actually. Please add a single test, similar to [this one](https://github.com/qiskit-community/ffsim/blob/561052c6486e7daf7f828a8d4096eff99f26cc33/tests/python/operators/common_operators_test.py#L21) for `number_operator`. The test should compare the dictionary of the operator with one constructed explicitly without any for loop logic. Then, you can revert these other tests back to how they were originally, where the only thing they did was check the eigenvalue. That gives a better logical separation between what the tests are doing. For good measure, you can rename the tests by appending `_eigenvalue` to the test name and mention it in the docstring, e.g.
```python
def test_non_interacting_fermi_hubbard_1d_eigenvalue():
    """Test non-interacting 1D Fermi-Hubbard model eigenvalues."""
```